interfaces/repo.go

```diff
@@ -754,7 +774,12 @@ func isAutoConnectCandidate(plug *Plug, slot *Slot) bool {
 	}
 
 	// OS snap auto connect candidates
-	if slot.Snap.Type == snap.TypeOS && slot.Interface == plug.Interface {
+	if slot.Snap.Type == snap.TypeOS {
+		return true
+	}
+
+	// FIXME: remove once we have assertions that provide this feature
+	if isSpecialAutoConnectSnap(plug.Snap) {
 		return true
 	}
```

Comment:
It's weird that this check is both in isAutoConnectCandidate() and AutoConnectCandidates() and it makes me wonder why the r.ifaces[plug.Interface].AutoConnect() check is not done in AutoConnectCandidates(). That said, this is temporary code until snap declarations are in place, and people are actively working on landing that now, so I don't see a compelling reason to complicate this landing with code-refactoring that will likely be refactored again with snap declarations.